LocalSettings.php

```diff
@@ -201,199 +213,344 @@
-$wgAddGroups['bureaucrat'] = array( 'bot', 'sysop', 'bureaucrat', 'interwiki-admin' );
-$wgRemoveGroups['bureaucrat'] = array( 'bot', 'sysop', 'bureaucrat', 'interwiki-admin' );
-$wgGroupsRemoveFromSelf['sysop'] = array( 'sysop' );
-$wgRestrictionLevels = array( '', 'user', 'bureaucrat', 'sysop', 'autoconfirmed', 'steward' );
-$wgRestrictionTypes = array( 'create', 'edit', 'move', 'upload', 'delete', 'protect' );
-$wgCascadingRestrictionLevels = array( 'bureaucrat', 'sysop', 'steward' );
-$wgSemiprotectedRestrictionLevels = array( 'user', 'autoconfirmed', );
+
+// Permission assignments
+/// Administrators
+$wgAddGroups['sysop'] = array(
+	'autopatrol',
+	'autoreview',
+	'blockedfromchat',
+	'chatmod',
+	'confirmed',
+	'editor',
+	'patroller',
+	'researcher',
+	'reviewer',
+	'translateadmin'
+);
+$wgRemoveGroups['sysop'] = array(
+	'autopatrol',
+	'autoreview',
+	'blockedfromchat',
+	'chatmod',
+	'confirmed',
+	'editor',
+	'patroller',
+	'researcher',
+	'reviewer',
+	'translateadmin'
+);
+
+/// Bureaucrats
+$wgAddGroups['bureaucrat'] = array(
+	'autoconfirmed',
+	'autopatrol',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+$wgRemoveGroups['bureaucrat'] = array(
+	'autoconfirmed',
+	'autopatrol',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+
+/// Stewards
+$wgAddGroups['steward'] = array(
+	'autoconfirmed',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'steward',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+$wgRemoveGroups['steward'] = array(
+	'autoconfirmed',
+	'bureaucrat',
+	'confirmed',
+	'patroller',
+	'suppress',
+	'sysop',
+	'translateadmin'
+);
+
+/// System administrators
+$wgAddGroups['sysadmin'] = true;
+$wgRemoveGroups['sysadmin'] = true;
+
+// Restriction settings
+$wgRestrictionLevels = array(
+	'*',
+	'user',
+	'autoconfirmed',
+	'autoreview',
+	'reviewer',
+	'sysop',
+	'bureaucrat',
+	'steward',
+	'sysadmin'
+);
+
+$wgRestrictionTypes = array(
+	'create',
+	'edit',
+	'move',
+	'upload',
+	'delete',
+	'bigdelete',
+	'purge',
+	'protect'
+);
+
+$wgCascadingRestrictionLevels = array(
+	'sysop',
+	'bureaucrat',
+	'steward',
+	'sysadmin'
+);
+
+$wgSemiprotectedRestrictionLevels = array(
+	'user',
+	'autoconfirmed',
+	'translateadmin'
+);
+
+// Unsets
 unset( $wgGroupPermissions['staff'] );
+
 unset( $wgGroupPermissions['editor'] );
+
 $wgExtensionFunctions[] = function() use ( &$wgGroupPermissions ) {
-    unset( $wgGroupPermissions['staff'] );
-    unset( $wgGroupPermissions['editor'] );
+    unset( $wgGroupPermissions['staff']
+);
+    unset( $wgGroupPermissions['editor']
+);
+
 };
-$wgShowExceptionDetails = true;
 
-$wgGrantPermissions['editprotected']['bureaucrat'] = true;
+// RC fed settings
+$wgRCFeeds['irc'] = [
+	'uri' =>
+	'udp://129.153.11.72:9872',
+	'formatter' => IRCColourfulRCFeedFormatter::class,
+	'add_interwiki_prefix' => false,
+	'omit_bots' => true,
+];
+
+// Other settings
 
-// Non-Steward suppressors
```

Comment:
why is this group removed?

Reply:
Superfluous! One does not need a role that can simply give and take away a more powerful role! Then one simply assigns the more powerful role directly

Reply:
That is against consensus. The consensus for NSSs was to allow certain people to add themselves suppressor temporarily using the non-stewardsuppress group, not assign it directly.

Reply:
What's the point of having a group only temporarily? Would it be better to apply for it permanently? And who is preventing them from having the group permanently?